scripts/coccinelle/bit-macro.cocci

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-3.0-only
+// Replace shifts with BIT macro
+
+@@ expression val; @@
+-(1 << val)
++BIT(val)
+
+@@ expression val; @@
+-(1U << val)
++BIT(val)
+
+@@ expression val; @@
+-1 << val
```

Comment:
This one seems potentially dangerous, it could over-match

Reply:
I'm not entirely sure how it matches, but it seems to match `1` exactly, as `0b11 << N` values were not modified.